python/llm/dev/benchmark/all-in-one/run.py

```diff
@@ -930,21 +990,62 @@ def run_transformer_autocast_bf16( repo_id,
+        while(flag):
+            mem_usage = line.split()[index]
+            print("-" * 100)
+            print("This is the entire output of the top command:\n", mem_line)
+            print("This is the extracted memory usage:\n", mem_usage)
+            print("-" * 100)
+            user_input = input("Please check if the memory usage is correct([y]/n/exit): ").strip().lower()
+            if user_input == "y" or user_input == "":
+                flag = False
+                print("Continuing with the process...")
+            elif user_input == "n":
+                try:
+                    index_input = input("Please type in an integer to update the index:")
+                    index = int(index_input)
+                except ValueError:
+                    print("Please type in an integer!")
+            else:
+                exit()
+
     import pandas as pd
     for api in conf.test_api:
         for model in conf.repo_id:
-            in_out_pairs = conf['in_out_pairs'].copy()
-            if excludes:
-                for in_out in conf['in_out_pairs']:
-                    model_id_input = model + ':' + in_out.split('-')[0]
-                    if model_id_input in excludes:
-                        in_out_pairs.remove(in_out)
-            run_model(model, api, in_out_pairs, conf['local_model_hub'], conf['warm_up'], conf['num_trials'], conf['num_beams'],
-                      conf['low_bit'], conf['cpu_embedding'], conf['batch_size'])
+            in_out_pairs_ = conf['in_out_pairs'].copy()
+            for in_out_pairs in in_out_pairs_:
+                in_out_pairs = [in_out_pairs]
+                if excludes:
+                    for in_out in conf['in_out_pairs']:
+                        model_id_input = model + ':' + in_out.split('-')[0]
+                        if model_id_input in excludes:
+                            in_out_pairs.remove(in_out)
+                if conf['plot_memory_vs_time_graph'] == True:
+                    run_model(model, api, in_out_pairs, conf['local_model_hub'], conf['warm_up'], conf['num_trials'], conf['num_beams'],
+                            conf['low_bit'], conf['cpu_embedding'], conf['batch_size'], index=index, plot_memory_vs_time_graph=True)
+                else:
+                    run_model(model, api, in_out_pairs, conf['local_model_hub'], conf['warm_up'], conf['num_trials'], conf['num_beams'],
+                            conf['low_bit'], conf['cpu_embedding'], conf['batch_size'], plot_memory_vs_time_graph=False)
         df = pd.DataFrame(results, columns=['model', '1st token avg latency (ms)', '2+ avg latency (ms/token)', 'encoder time (ms)',
                                             'input/output tokens', 'actual input/output tokens', 'num_beams', 'low_bit', 'cpu_embedding', 
                                             'peak mem (GB)'])
 
-        df.to_csv(f'{current_dir}/{api}-results-{today}.csv')
+        df.to_csv(f'{current_dir}/output/{api}-results-{today}.csv')
```

Comment:
hi @Oscilloscope98 , this will change the location of our nightly test results right?

Reply:
Yes. And this may cause some problems